nncf/openvino/graph/nncf_graph_builder.py

```diff
@@ -88,19 +89,28 @@ def _add_edges_to_nncf_graph(model: ov.Model, graph: NNCFGraph) -> None:
         for op in model.get_ops():
             in_node_id = graph.get_node_by_name(op.get_friendly_name()).node_id
             for output_port_id, out in enumerate(op.outputs()):
+                node_vs_target_inputs = defaultdict(list)
                 for inp in out.get_target_inputs():
-                    out_node = inp.get_node()
+                    node_vs_target_inputs[inp.get_node()].append(inp)
+
+                for out_node, inputs in node_vs_target_inputs.items():
                     tensor_shape = list(out.partial_shape.get_max_shape())
                     output_node_id = graph.get_node_by_name(out_node.get_friendly_name()).node_id
                     ov_dtype = out.get_element_type().get_type_name()
                     nncf_dtype = GraphConverter.convert_to_nncf_dtype(ov_dtype)
+
+                    parallel_inputs_kwarg = {}
+                    if len(inputs) > 1:
+                        parallel_inputs_kwarg = {"parallel_input_port_ids": [inp.get_index() for inp in inputs]}
```

Comment:
Stop building `kwargs` dicts in your pull requests, this is not a refactoring-friendly, readability-friendly or type-checking-friendly solution. If params are optional, and shouldn't be used in a given case, set these to `None` values.

Reply:
Ok, I need to refactor `graph.add_edge_between_nncf_nodes` then

Reply:
Done